### ISSUE TITLE
resample_flux edge effects

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#77]
test_resample.py now includes a unitttest flagged as `expectedFailure` because it identifies an edge effect bug in `resample_flux`.  Note how the last resampled bin is way lower that it apparently should be.

```
import numpy as np
from desispec.interpolation import resample_flux

x = np.arange(0.0, 30)
y = np.sin(x/4)+1
xx = np.linspace(1, 29, 13)
yy = resample_flux(xx, x, y)

clf()
plot(x, y, 'b.-')
plot(xx, yy, 'r.-')
```

![screen shot 2015-10-01 at 3 38 26 pm](https://cloud.githubusercontent.com/assets/218471/10235584/7b63a6e2-6852-11e5-83ce-e62818c26ce4.png)

**Comment by julienguy** [comment]
This is not a bug but a feature of the routine resample_flux where, as written in the docstring,

The input is interpreted as the nodes positions and node values of
    a piece-wise linear function y(x) = sum_i y_i * f_i(x)
and
the output value is the average flux density in a bin
        flux_out(j) = int_{x>(x_{j-1}+x_j)/2}^{x<(x_j+x_{j+1})/2} y(x) dx /  0.5*(x_{j+1}+x_{j-1})

This is however an issue for some use of this routine.
For this purpose, I have added an option "extrapolate" which considers that the input flux density is the same as the first (last) input values for output x smaller (larger) than the input x range.
This is in branch "interpolate". When applied to the above case, this gives 
![figure_1](https://user-images.githubusercontent.com/5192160/27867366-bb6ad8cc-6199-11e7-9c6c-31d2fb44c7b0.png)

**Comment by moustakas** [comment]
Fixed in #415.
